modules/photons_control/device_finder.py

```diff
@@ -541,13 +576,86 @@ def location_id(self):
             return sb.NotSpecified
         return self.location.uuid
 
+    @property
+    def product_id(self):
+        if self.product is sb.NotSpecified:
+            return sb.NotSpecified
+        return self.product.pid
+
+    @property
+    def product_name(self):
+        if self.product_id is sb.NotSpecified:
+            return sb.NotSpecified
+        return self.product.friendly
+
+    @property
+    def product_type(self):
+        if self.product is not sb.NotSpecified:
+            return (
+                ProductType.LIGHT.value
+                if Products[VendorRegistry.LIFX, self.product.pid].cap.is_light
```

Comment:
```suggestion
                if self.cap.is_light
```

Reply:
can even remove the check on self.product

Reply:
and check if cap is unknown for returning ProductType.UNKNOWN rather than unspecified

Reply:
So I did that initially, but it required waiting for enough response which meant putting logic in weird places. So I moved the logic to determine the best current value based off what we know from the packets we have thus far.

Reply:
but self.product should give back either the product or an instance of https://github.com/delfick/photons/blob/main/modules/photons_products/base.py#L228 (which we should make better, in a pre-PR) and self.cap should be based off self.product.